Add crystal-radix

marksiemers opened the pull request.

marksiemers committed:
Add crystal-radix

RX14 commented:
I said earlier, this should be in the `Crystal/crystal` folder as it shares so much: the dependencies and the ECR views.

RX14 commented:
You can have multiple tests in one `benchmark_config.json` files with different names, and different setup files. See how go does it with `default` and `prefork` https://github.com/marksiemers/FrameworkBenchmarks/blob/master/frameworks/Go/go-std/benchmark_config.json

marksiemers commented on frameworks/Crystal/crystal/benchmark_config.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "framework": "crystal-raw",
+  "framework": "crystal",
```
Will this mess up reporting?

marksiemers commented:
@RX14 - I moved the crystal-radix test inside the crystal folder. It means that radix was added as a dependency in the shared `shards.yml` file. I'm assuming that won't affect the `crystal-raw` tests.

I also changed the name of the framework to `crystal` from `crystal-raw`. If that is going to mess up reporting and comparisons, I can change it back.

@drujensen @elorest @faustinoaq @eliasjpr - Any other tests that we want to add before I open a PR with TechEmpower? To pinpoint the problem we may need quite a few tests - structs, classes, Procs, etc.

Not that my local testing means anything, but adding the radix tree dropped the performance to about 92% of `crystal-raw` while `kemal` and `amber` typically drop to 50-60% (locally).

drujensen reviewed (approved):
looks good.  not sure what else we can test.  thanks for taking the initiative on this.
